LunrCore/Index.cs

```diff
@@ -313,7 +313,10 @@ await foreach (Result result in Search(queryString, CultureInfo.CurrentCulture,
                     {
                         foreach (string field in clause.Fields)
                         {
-                            requiredMatches.Add(field, Set<string>.Empty);
+                            if (!requiredMatches.TryGetValue(field, out var value))
```

Comment:
I'm not sure this is the correct fix. First, if we wanted to just add it or do nothing if it's already there, here we'd be doing a double lookup which is wasteful. Pretty sure you could do `requiredMatches[field] = Set<string>.Empty`. More importantly, can you add a test that demonstrates the problem and that's fixed by your change?